### PR TITLE
Warn when opening a truncated core file

### PR DESCRIPTION
**Pull request by keithc-ca**
This is a replay of #9199 with a second commit that fixes the compile problem.

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile aix,plinux,win jdk8

File: debugtools/DDR_VM/generate.xml [hunk 1]
```diff
@@ -172,13 +172,11 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
 				<shellscript shell="bash" dir="${TMP_DIR}">
-					mkdir ${TMP_DIR}/debugtools
-					cp -af ${WORKSPACE}/debugtools/DDR_* ${TMP_DIR}/debugtools
-					cp -af ${WORKSPACE}/jcl ${TMP_DIR}
-					cp -af ${WORKSPACE}/../build/RAS_Binaries/*.jar ${TMP_DIR}/RAS_Binaries
-					mkdir ${TMP_DIR}/sourcetools/
-					cp -af ${WORKSPACE}/sourcetools/lib ${TMP_DIR}/sourcetools/
-					cp -af ${WORKSPACE}/sourcetools/*.mk ${TMP_DIR}/sourcetools/
+					mkdir -p debugtools/ RAS_Binaries/ sourcetools/
+					cp -af ${WORKSPACE}/jcl ./
```
Comment (pshipton) [inline debugtools/DDR_VM/generate.xml:176]: This appears to be a change in behavior, as the original code copied to ${TMP_DIR}. Why the change?

Comment (keithc-ca) [reply debugtools/DDR_VM/generate.xml:176]: It's not a change in behavior: the commands are executed in the directory named `${TMP_DIR}`.

Comment (pshipton) [reply debugtools/DDR_VM/generate.xml:176]: Ok, but originally the script wasn't dependent on the cd, and now it is. This doesn't seem like an improvement.

Comment (keithc-ca) [reply debugtools/DDR_VM/generate.xml:176]: I can add seven uses of `${TMP_DIR}` if you feel strongly, but, in my opinion, _that_ would not be an improvement.

Comment (pshipton) [reply debugtools/DDR_VM/generate.xml:176]: Ok, I see what you mean.
